src/jit/gentree.cpp

```diff
@@ -10538,10 +10544,12 @@ void Compiler::gtDispRegVal(GenTree* tree)
         }
     }
 
-    if (tree->gtFlags & GTF_REG_VAL)
+#ifdef CODEGEN_LEGACY
+    if (tree->InReg())
```

Comment:
Ouch - yes. Believe it or not, I had actually tested LEGACY_BACKEND on an earlier version of these changes, but that was before I'd change `InReg()` from always returning false for RyuJIT backend, to eliminating it.